src/lib/s2k.c

```diff
@@ -85,7 +85,7 @@ pgp_s2k_round_iterations(size_t iterations)
 uint8_t
 pgp_s2k_encode_iterations(size_t iterations)
 {
-    for (uint8_t c = 0; c <= 255; ++c) {
+    for (uint8_t c = 0; c < 255; ++c) {
```

Comment:
@catap , @ronaldtse : The better fix would change the datatype of the iterator from uint8_t to uint16_t and change the boundary value to 256. 
`for (uint16_t c = 0; c < 256; ++c)`